iis2iclx_reg.c

```diff
@@ -6114,21 +6114,23 @@ int32_t iis2iclx_mlc_out_get(const stmdev_ctx_t *ctx, uint8_t *buff)
   *
   */
 int32_t iis2iclx_sh_read_data_raw_get(const stmdev_ctx_t *ctx,
-                                      iis2iclx_emb_sh_read_t *val)
+                                      iis2iclx_emb_sh_read_t *val, uint16_t len)
 {
   int32_t ret;
 
-  ret = iis2iclx_mem_bank_set(ctx, IIS2ICLX_SENSOR_HUB_BANK);
+  if (len < 19) {

```

Comment:
I prefer that you exit here. If you want to be MISRA compliant and have only one point of exit then just do:

   ```
   if (ret < 19)
   {
      ret = -1; (or -EINVAL)
      got exit;
   }
```

And pls add also a comment before the test ;)

Reply:
@albezanc it is almost perfect. Only one note on the ```ret = -1;``` See below